multiversx_sdk/network_providers/proxy_network_provider.py

```diff
@@ -102,16 +109,20 @@ def get_tx() -> Dict[str, Any]:
 
         status_task = None
         with ThreadPoolExecutor(max_workers=2) as executor:
-            if with_process_status:
-                status_task = executor.submit(get_process_status)
+            try:
+                if with_process_status:
+                    status_task = executor.submit(get_process_status)
 
-            tx_task = executor.submit(get_tx)
+                tx_task = executor.submit(get_tx)
 
-        process_status = status_task.result() if status_task else None
-        tx = tx_task.result()
-        transaction = TransactionOnNetwork.from_proxy_http_response(tx_hash, tx, process_status)
+                process_status = status_task.result(timeout=5) if status_task else None
+                tx = tx_task.result(timeout=5)
```

Comment:
5 is a constant here, should be defined somewhere in `self.config`.

Reply:
As discussed, we've removed the timeout for the threads since we have a timeout for the requests.